File: packages/rust/tests/fixtures/integration_lint/typescript/no_first_party_mock/red/charge.test.ts
import { describe, expect, it, vi } from 'vitest';

import { charge } from '../src/charge';

// VIOLATION: an integration test must run first-party code for real, so mocking
// a first-party collaborator (a relative import) is forbidden.
vi.mock('../src/ledger', () => ({
  record: vi.fn(),
}));

// Allowed: the external payment gateway is a third-party package.
vi.mock('stripe');

describe('charge (integration)', () => {
  it('records the charge', async () => {
    await charge({ amount: 100 });
    expect(true).toBe(true);
  });
});

File: packages/rust/tests/fixtures/integration_lint/python/no_first_party_patch/red/tests/integration/charge_test.py
# Red fixture for the Python integration-isolation rule (#42, `no-first-party-patch`).
# An integration test runs first-party code for real, so patching a first-party
# target is the violation. The patch lives in a fixture (so it does NOT also trip
# `no-inline-patch`) — isolating the new rule.
from unittest.mock import patch

import pytest

from myproject.charge import charge


@pytest.fixture
def fake_ledger():
    # VIOLATION: `myproject.ledger` is first-party — an integration test must run
    # it for real. Mock the outside world, not our own code.
    with patch("myproject.ledger.record") as mock:
        yield mock


def describe_charge():
    def it_records_the_charge(fake_ledger):
        charge(amount=100)
        assert fake_ledger.called
